src/ssp.cpp

```diff
@@ -204,30 +118,126 @@ void InitQuad(SSP_INIT_ARGS)
     ssp->Nz = ssp->NPts;
 }
 
-void InitializeSSP(SSP_INIT_ARGS)
+/**
+* Update the SSP parameters. Safe to call multiple times with flags.
+* Be sure to flag ssp->dirty if you change the SSP externally.
+* TODO: You have to provide FileRoot if you use a 2D SSP.
+*/
+void UpdateSSP(vec2 x, real Depth, real freq, const real& fT, SSPStructure* ssp, 
+    PrintFileEmu& PRTFile, const AttenInfo* atten, std::string FileRoot)
```

Comment:
Where is FileRoot needed in this function?

Reply:
InitQuad uses it to read in the 2D SSP. I give it a default value for now that can be skipped, but it's on the list to handle cleaner.

Reply:
Ah sorry I missed that. Since it is used and would be required for certain env file contents, it probably shouldn't have a default value--nowhere in the code should call it without specifying its real value.